src/esp/metadata/attributes/LightLayoutAttributes.cpp

```diff
@@ -39,19 +39,57 @@ LightLayoutAttributes::LightLayoutAttributes(const std::string& handle)
   // set default scaling for positive and negative intensities to 1.0
   setPositiveIntensityScale(1.0);
   setNegativeIntensityScale(1.0);
+  // get ref to internal subconfig for light instances
+  lightInstConfig_ = editSubconfig<Configuration>("light_instances");
+}
+LightLayoutAttributes::LightLayoutAttributes(const LightLayoutAttributes& otr)
+    : AbstractAttributes(otr), availableLightIDs_(otr.availableLightIDs_) {
+  lightInstConfig_ = editSubconfig<Configuration>("light_instances");
+
+  copySubconfigIntoMe<LightInstanceAttributes>(otr.lightInstConfig_,
+                                               lightInstConfig_);
+}
+LightLayoutAttributes::LightLayoutAttributes(
+    LightLayoutAttributes&& otr) noexcept
+    : AbstractAttributes(std::move(static_cast<AbstractAttributes>(otr))),
+      availableLightIDs_(std::move(otr.availableLightIDs_)) {
+  lightInstConfig_ = editSubconfig<Configuration>("light_instances");
+}
+
+LightLayoutAttributes& LightLayoutAttributes::operator=(
+    const LightLayoutAttributes& otr) {
+  if (this != &otr) {
+    this->AbstractAttributes::operator=(otr);
+    // point to our own light instance config and available ids
+    availableLightIDs_ = otr.availableLightIDs_;
+    lightInstConfig_ = editSubconfig<Configuration>("light_instances");
+    copySubconfigIntoMe<LightInstanceAttributes>(otr.lightInstConfig_,
+                                                 lightInstConfig_);
+  }
+  return *this;
+}
+LightLayoutAttributes& LightLayoutAttributes::operator=(
+    LightLayoutAttributes&& otr) noexcept {
+  // point to our own light instance config and available ids
+  availableLightIDs_ = std::move(otr.availableLightIDs_);
+  this->AbstractAttributes::operator=(
+      std::move(static_cast<AbstractAttributes>(otr)));
+  lightInstConfig_ = editSubconfig<Configuration>("light_instances");
+  return *this;
 }
```

Comment:
I find it a bit problematic to have to manually implement copy/move constructors and assignments for the subclasses. Is it really essential to have the `lightInstConfig_` member manually cached here? What is in it that can't be implicitly managed by the base class already? I thought the base takes care of such hierarchic configuration on its own, and so when there's a config group named `light_instances` then a copy constructor does the right thing and copies it as well?

Same question about `SceneAttributes`.